core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsAddressCache.java

```diff
@@ -0,0 +1,42 @@
+package org.infinispan.remoting.transport.jgroups;
+
+import org.infinispan.commons.equivalence.AnyEquivalence;
+import org.infinispan.commons.util.concurrent.jdk8backported.EquivalentConcurrentHashMapV8;
+import org.jgroups.Address;
+import org.jgroups.util.ExtendedUUID;
+import org.jgroups.util.UUID;
+
+/**
+ * Cache JGroupsAddress instances
+ *
+ * @author Dan Berindei
+ * @since 7.0
+ */
+public class JGroupsAddressCache {
+   private static final EquivalentConcurrentHashMapV8<Address, JGroupsAddress> addressCache =
+         new EquivalentConcurrentHashMapV8<>(AnyEquivalence.getInstance(), AnyEquivalence.getInstance());
+
+   static org.infinispan.remoting.transport.Address fromJGroupsAddress(final org.jgroups.Address addr) {
```

Comment:
By the way did you see a big improvement in overall message size with the cache ?